src/main/java/teammates/ui/template/InstructorFeedbackResultsGroupByQuestionPanel.java

```diff
@@ -0,0 +1,80 @@
+package teammates.ui.template;
+
+import java.util.List;
+
+import teammates.common.util.FieldValidator;
+
+
+/**
+ * Data model for the giver panel in InstructorFeedbackResults for Giver > Question > Recipient,
+ * and for the recipient panel in Recipient > Question > Giver
+ * 
+ *
+ */
+public class InstructorFeedbackResultsGroupByQuestionPanel extends InstructorResultsParticipantPanel {
+    
+    List<InstructorResultsQuestionTable> questionTables;
+    
+    boolean isEmailValid;
+    
+    public InstructorFeedbackResultsGroupByQuestionPanel(List<InstructorResultsQuestionTable> questionTables,
+                                    String profilePictureLink, 
+                                    boolean isGiver, String participantIdentifier, String participantName,
+                                    InstructorResultsModerationButton moderationButton, 
+                                    boolean isModerationButtonDisplayed) {
+        setParticipantIdentifier(participantIdentifier);
+        setName(participantName);
+        setGiver(isGiver);
+        
+        boolean isEmailValid = new FieldValidator()
+                                       .getInvalidityInfo(FieldValidator.FieldType.EMAIL, participantIdentifier).isEmpty();
+        setEmailValid(isEmailValid);
+        this.profilePictureLink = profilePictureLink;
+        
+        this.questionTables = questionTables;
+        
+        setModerationButton(moderationButton);
+        setModerationButtonDisplayed(isModerationButtonDisplayed);
+        
+        setHasResponses(true);
```

Comment:
`moderationButton` is accessible from this class, so it's possible to set it directly.
Not sure if that's intended, though.
It should be fine setting the variables to be accessible by its child classes.